aiohttp/connector.py

```diff
@@ -770,6 +774,9 @@ def __init__(
         self._throttle_dns_events: Dict[Tuple[str, int], EventResultOrError] = {}
         self._family = family
         self._local_addr = local_addr
+        self._network_interface = network_interface
+        if self._network_interface and not hasattr(socket, "SO_BINDTODEVICE"):
```

Comment:
I'd put the feature-detection into a dedicated *named* module-level constant like
```python
SOCKET_SUPPORTS_BINDING_TO_DEVICE = hasattr(socket, "SO_BINDTODEVICE")
```
Maybe even store that in the `aiohttp.helpers` module.